### PR TITLE
Feature/pagerank

### PR DESCRIPTION
**Pull request by brianignacio5**
#12 Please do a code review. The test object are huge i know, but i need something small but representative of a web graph to make the page rank idea.

I think is a good third step (or so...) please do a code review when you can.

The only missing thing is to modify the uuid for raw_filename  to use this page rank score in rank(). Didn't touch this part since it was  being worked in another branch (feature/conjunctive-ranking). Maybe we can decide in a middle man (a file with both values) to save uuid and filename to avoid many modifications, since many features depend on uuid.

Well, we talk later.

### CODE REVIEW

File: test/test_page_ranker.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+
+	for i, url in enumerate(urls):
+		if os.path.exists(files[i]):
+			pass
+		else:
+			webpage = get_site(url)
+			with open(files[i], 'w') as f:
+				f.write(webpage.text)
+
+	test_link_database = {'author_w_Wilbon:Anthony_D=.html': ['journal_ijea_ijea5.html'],
```
Comment (moritzschaefer) [inline test/test_page_ranker.py]: Why don't just take 5 or so???

File: bin/tsg_pagerank [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+logging.basicConfig(level=logging.INFO)
+
+
+def main():
+    print('Do you want to delete pagerank file? y/N')
+    if input() != 'y':
+        sys.exit(0)
+
+    # get page rank
+    page_rank_scores = test_calc_page_rank(RAW_DIR)
```
Comment (moritzschaefer) [inline bin/tsg_pagerank]: why is it called test_ btw?

File: bin/tsg_pagerank [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+    # convert to dataframe
+    df = pd.DataFrame(page_rank_scores)
+
+    df.to_json(PAGERANK_PATH)
```
Comment (moritzschaefer) [inline bin/tsg_pagerank]: we should use csv right?

File: bin/tsg_pagerank (hunk 2 above)
Comment (brianignacio5) [inline bin/tsg_pagerank]: Sorry bad copy/paste.

File: bin/tsg_pagerank (hunk 3 above)
Comment (brianignacio5) [inline bin/tsg_pagerank]: What do you prefer?

File: bin/tsg_pagerank (hunk 2 above)
Comment (moritzschaefer) [inline bin/tsg_pagerank]: copy paste is evil!!

File: bin/tsg_pagerank (hunk 3 above)
Comment (moritzschaefer) [inline bin/tsg_pagerank]: CSV

File: test/test_page_ranker.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import os
+import tsg.indexer.page_rank as page_rank
+
+RAW_TEST_DIRECTORY = os.path.abspath('test') + "/files/pagerank/"
```
Comment (brianignacio5) [inline test/test_page_ranker.py]: You need to run the test from .../the-search-engine/ for this to work.

**Pull request merged**